### PR TITLE
M3-2699 Fix: Monthly Network Transfer (Linode Summary)

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

Data in the Monthly Network Transfer panel on Linode Summary is currently incorrect. Network traffic from the last 30 days (not just the current month) is being used, and the average _rates_ are being shown – not the actual values

To fix this, only stat readings from the current month are summed. The sum is then multiplied by 7200 (the number of seconds in 2 hours, since the stat readings are two-hour averages of bits/second).

Screen Shot 2019-04-23 at 11 47 34 AM
Screen Shot 2019-04-23 at 6 12 16 PM

I added a new utility function, `readableBytes`, since the library we are currently using to do this, `pretty-bytes`, uses base-10 (e.g. kilobyte instead of kibibyte).

## Type of Change
- Bug fix ('fix', 'repair', 'bug')

## Applicable E2E Tests
None

## Note to Reviewers

This is kind of a pain to test. The best way is to compare with Classic Manager, but since we get two-hour averages from the API, the data isn't going to be totally in sync.

I suggest creating a Linode, driving up the traffic, turning it off, waiting a few hours, and comparing with Classic.

### CODE REVIEW

File: src/utilities/readableBytes.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  } else if (result < 10) {
+    decimalPlaces = 2;
+  } else if (result < 100) {
+    decimalPlaces = 1;
+  } else {
+    decimalPlaces = 0;
+  }
+
+  const value = parseFloat(result.toFixed(decimalPlaces));
+
```
Comment (martinmckenna) [inline src/utilities/readableBytes.ts]: lets combine this with `convertMegaBytesTo.ts` file

File: src/features/linodes/LinodesDetail/LinodeSummary/LinodeNetSummary.tsx [hunk 2]
```diff
@@ -8,23 +8,23 @@ import { WithStyles } from 'src/components/core/styles';
...
+import { MapState } from 'src/store/types';
+import { isRecent } from 'src/utilities/isRecent.ts';
+import { readableBytes } from 'src/utilities/readableBytes';
+import { getMonthlyNetworkTransferInBits } from 'src/utilities/statMetrics';
 
 interface Props {
   linodeId: number;
 }
 
 interface StateProps {
```
Comment (martinmckenna) [inline src/features/linodes/LinodesDetail/LinodeSummary/LinodeNetSummary.tsx]: _StateProps_ makes me think these are coming from Redux. Also these are included in `CombinedProps` ❓

File: src/utilities/readableBytes.ts (hunk 1 above)
Comment (johnwcallahan) [reply src/utilities/readableBytes.ts]: Do you have ideas for how to combine them? I want to keep the rounding rules and return object from this function. I can change `convertMegabytesTo` to use this function, but it's going to change decimal precision on NBs.


E.g.:
```
export const convertMegabytesTo = (data: number) => {
  // API v4 always returns nodebalancer transfer in MB, so we want to clean it up if it's too
  // big or too small
  const totalToBytes = data * 1024 * 1024; // convert the MB to Bytes

  return readableBytes(totalToBytes).formatted;
};
```

File: src/features/linodes/LinodesDetail/LinodeSummary/LinodeNetSummary.tsx (hunk 2 above)
Comment (johnwcallahan) [reply src/features/linodes/LinodesDetail/LinodeSummary/LinodeNetSummary.tsx]: Lol yeah just noticed this. Just pushed up a change.

File: src/utilities/readableBytes.ts (hunk 1 above)
Comment (martinmckenna) [reply src/utilities/readableBytes.ts]: oh no I just meant put both functions in one file lol

**Pull request closed without merging**